webapp/packages/plugin-data-spreadsheet-new/src/DataGrid/TableColumnHeader/TableColumnHeader-new.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * cloudbeaver - Cloud Database Manager
+ * Copyright (C) 2020 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+
+import { computed } from 'mobx';
+import { observer } from 'mobx-react-lite';
+import { useCallback, useContext, useMemo } from 'react';
+import type { HeaderRendererProps } from 'react-data-grid';
+import styled, { css, use } from 'reshadow';
+
+import { StaticImage, Icon } from '@cloudbeaver/core-blocks';
+import type { SqlResultSet } from '@cloudbeaver/core-sdk';
+import type { SortMode } from '@cloudbeaver/plugin-data-viewer';
+
+import { DataGridContext } from '../DataGridContext';
+import { DataGridSortingContext } from '../DataGridSorting/DataGridSortingContext';
+
+const headerStyles = css`
+  table-header {
+    display: flex;
+    align-items: center;
+    align-content: center;
+    width: 100%;
+  }
+  shrink-container {
+    display: flex;
+    align-items: center;
+    flex: 1 1 auto;
+    overflow: hidden;
+  }
+  icon {
+    height: 16px;
+  }
+  StaticImage {
+    height: 100%;
+  }
+  name {
+    margin-left: 8px;
+    font-weight: 400;
+    flex-grow: 1;
+  }
+  
+  sort-icon {
+    margin-left: 4px;
+    display: flex;
+    padding: 2px 4px;
+    flex-direction: column;
+    align-content: center;
+    align-items: center;
+    min-width: 20px;
+    box-sizing: border-box;
+    cursor: pointer;
+  }
+
+  sort-icon > Icon {
+    width: 8px;
+    fill: #cbcbcb;
+  }
+  sort-icon > Icon:last-child {
+    transform: scaleY(-1);
+  }
+  sort-icon > Icon[|active] {
+    fill: #338ECC;
+  }
+  sort-icon[|disabled] {
+    opacity: 0.7;
+    cursor: default;
+  }
+  sort-icon:hover > Icon {
+    width: 9px;
+  }
+  sort-icon[|disabled]:hover > Icon {
+    width: 8px;
+  }
```

Comment:
seems you just need to declare disabled styles after others and remove that override